### ISSUE TITLE
Extended Enum caption order is different to the Enum value when used as an option field.

### ISSUE THREAD

**Issue by abhishekv290691** [microsoft/AL#5651]
**Describe the bug**
When an Enum is extended by multiple apps and used as a field in a page, the enum caption is different to the enum value being  stored.

**To Reproduce**
Steps and to reproduce the behavior:

1. Create a "Base App" with an extensible enum.

```
enum 50100 BaseAppEnum
{
    Extensible = true;
    value(0; none)
    {
        Caption = 'None';
    }
}
```

2. Create a Table which uses this extensible enum and Page to display the fields.

```
table 50100 BaseAppTable
{
    Caption = 'BaseAppTable';
    DataClassification = ToBeClassified;
    fields
    {
        field(50100; Id; Code[20])
        {
            Caption = 'Id';
            DataClassification = CustomerContent;
        }
        field(50101; BaseAppEnum; Enum BaseAppEnum)
        {
            Caption = 'BaseAppEnum';
            DataClassification = CustomerContent;
        }
    }
    keys
    {
        key(PK; Id)
        {
            Clustered = true;
        }
    }
    trigger OnModify()
    begin
        If Rec.BaseAppEnum <> xRec.BaseAppEnum then begin
            Message(Format(BaseAppEnum));
        end;
    end;
}
```

2. Create  2 extension apps which extend the enum on the base app.

Extension App 1

```
enumextension 50103 EnumExt2 extends BaseAppEnum
{
    value(1; EnumValue1)
    {
        Caption = 'Enum1';
    }
}
```

Extension App 2

```
enumextension 50104 EnumExt2 extends BaseAppEnum
{
    value(2; EnumValue2)
    {
        Caption = 'Enum2';
    }
}
```

3. Change the enum value in the page  and check the value being stored

**Expected behavior**
The Enum captions should match with the actual enum value.

**Screenshots**
Enum options
![image](https://user-images.githubusercontent.com/49511747/74303390-b5066e80-4dad-11ea-8e46-6b06573397c5.png)

After selecting an Enum 
![image](https://user-images.githubusercontent.com/49511747/74303449-f8f97380-4dad-11ea-9b77-7f04e77dbb4a.png)

4. Additional Info 
The caption sorting order may be related to the appId.
The app Id for the second extension is smaller. 

Extension 1 app ID : dbc51127-b453-44ef-b5e4-94fa12cb7cbb
Extension 2 app ID : ab9e43ca-3f11-4c75-8af2-201593fd5cf2

**5. Versions:**

 - AL Language:  4.0.0
 - Business Central: Platform 14.0.37683.0 + Application 37991 (AU Business Central 14.5)

**Comment by AndreasGloeckner** [comment]
Bug is still in version 
 - AL Language: 4.0.206840
 - Business Central: DE Business Central 15.2 (Platform 15.0.39461.39737 + Application 15.2.39558.39796)

**Comment by esbenk** [comment]
This is fixed in the upcoming 15.3 and in already released in Cumulative Update 09 for Microsoft Dynamics 365 Business (14.10) https://support.microsoft.com/help/4539529
